Assets/MixedRealityToolkit/_Core/Definitions/Utilities/InscribedRectangle.cs

```diff
@@ -0,0 +1,570 @@
+                    else
+                    {
+                        // Update the bottom collision point
+                        if (!EdgeUtilities.IsValidPoint(bottomCollisionPoint) ||
+                            (Vector2.SqrMagnitude(point - verticalIntersectionPoint) < Vector2.SqrMagnitude(point - bottomCollisionPoint)))
+                        {
+                            bottomCollisionPoint = verticalIntersectionPoint;
+                        }
+                    }
+                }
+
+                // Look for a horizontal collision
+                Vector2 horizontalIntersection = EdgeUtilities.GetIntersectionPoint(geometryEdges[i], horizontalLine);
+                if (EdgeUtilities.IsValidPoint(horizontalIntersection))
+                {
+                    // Is this intersection to the left or the right of the point?
+                    if (RotatePoint(horizontalIntersection, point, -angleRadians).x < point.x)
+                    {
+                        // Update the left collision point
+                        if (!EdgeUtilities.IsValidPoint(leftCollisionPoint) ||
+                            (Vector2.SqrMagnitude(point - horizontalIntersection) < Vector2.SqrMagnitude(point - leftCollisionPoint)))
+                        {
+                            leftCollisionPoint = horizontalIntersection;
+                        }
+                    }
+                    else
+                    {
+                        // Update the right collision point
+                        if (!EdgeUtilities.IsValidPoint(rightCollisionPoint) ||
+                            (Vector2.SqrMagnitude(point - horizontalIntersection) < Vector2.SqrMagnitude(point - rightCollisionPoint)))
+                        {
+                            rightCollisionPoint = horizontalIntersection;
+                        }
+                    }
+                }
+            }
+
+            // Each corner of the rectangle must intersect with the geometry.
+            if (!EdgeUtilities.IsValidPoint(topCollisionPoint) || 
+                !EdgeUtilities.IsValidPoint(bottomCollisionPoint) ||
+                !EdgeUtilities.IsValidPoint(leftCollisionPoint) || 
+                !EdgeUtilities.IsValidPoint(rightCollisionPoint))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Rotate a two dimensiopnal point about another point by the specified angle.
+        /// </summary>
+        /// <param name="point">The point to be rotated.</param>
+        /// <param name="origin">The point about which the rotation is to occur.</param>
+        /// <param name="angleRadians">The angle for the rotation, in radians</param>
+        /// <returns>
+        /// The coordinates of the rotated point.
+        /// </returns>
+        private Vector2 RotatePoint(Vector2 point, Vector2 origin, float angleRadians)
+        {
+            if (0.0f == angleRadians)
```

Comment:
`if (angleRadians.Equals(0f))` for floating point precision

Reply:
we should update the june18_dev code then too